### ISSUE TITLE
[Feature request] Device tracker popup 

### ISSUE THREAD

**Issue by alphasixtyfive** [resoai/TileBoard#576]
Hi guys!

I was thinking, perhaps it would be a good idea to open a popup with a big map when you click on the device tracker tile and if one has coordinates. What do you think?

**Comment by akloeckner** [comment]
Sounds like a neat enhancement. Are you thinking, just a fullscreen `DEVICE_TRACKER` tile with fixed `zoom` setting (=should be easy with popups right now)? Or an actual embedded map (= not sure how that would be done)?

**Comment by alphasixtyfive** [comment]
I think a fixed zoom full screen would be enough for everyone.

**Comment by akloeckner** [comment]
How about something like this?
```js
                  {
                     type: TYPES.DEVICE_TRACKER,
                     id: 'device_tracker.ak_honor',
                     position: [3,0],
                     map: 'mapbox',
                     zoomLevels: [9,13],
                     slidesDelay: 2,
                     action: function(item, entity) {
                        this.$scope.openPopup(item, entity);
                     },
                     popup: {
                        classes: ['-popup-landscape'],
                        styles: {},
                        items: [{
                           type: TYPES.DEVICE_TRACKER,
                           id: 'device_tracker.ak_honor',
                           map: 'mapbox',
                           zoomLevels: [6],
                           position: [0,0],
                           width: 5, height: 4,
                           classes: ['-item-fullsize'],
                           customStyles: {
                              width: null,
                              height: null,
                              top: null,
                              left: null,
                           },
                           action: false,
                           secondaryAction: false,
                        }]
                     }                     
                  }
```

We'd have to change this function, to use `cacheInItem`, then:
https://github.com/resoai/TileBoard/blob/a057d2a146492eb62faaa9d819ef723efcaa42cb/scripts/controllers/main.js#L320
(Probably should anyways, in case someone wants multiple different tiles of the same tracker...)

Probably, we'll also want to add some magic to retrieve the actual fullsize tile size, here:
https://github.com/resoai/TileBoard/blob/a057d2a146492eb62faaa9d819ef723efcaa42cb/scripts/controllers/main.js#L195
(Currently, the bogus `width` and `height` determine the image resolution.)

**Comment by alphasixtyfive** [comment]
Yeah, width, height, position and styles for all popups... perhaps we could could have all of them as defaults, but great otherwise!

**Comment by akloeckner** [comment]
> perhaps we could could have all of them as defaults

:-) we definitively should.

I think, it should be relatively easy to turn this stub into a PR. Unfortunately, I'm caught up in other projects currently. But if you are willing to start developing a feature, I'll happily offer test and feedback! 

I would consider the following locations as blueprints for the new popup:

Default action: https://github.com/resoai/TileBoard/blob/a057d2a146492eb62faaa9d819ef723efcaa42cb/scripts/globals/constants.js#L435

Opening function: https://github.com/resoai/TileBoard/blob/a057d2a146492eb62faaa9d819ef723efcaa42cb/scripts/controllers/main.js#L1615

Popup definition: https://github.com/resoai/TileBoard/blob/a057d2a146492eb62faaa9d819ef723efcaa42cb/scripts/globals/constants.js#L300

**Comment by alphasixtyfive** [comment]
Unfortunately I barely have any time myself and I did not even setup new env after changing project structure.

**Comment by akloeckner** [comment]
Oh, and that was quite awhile ago... let's just stash the idea for latter then...
